core/trino-main/src/main/java/io/trino/execution/FutureStateChange.java

```diff
@@ -34,6 +34,7 @@ public class FutureStateChange<T>
     @GuardedBy("listeners")
     private final Set<SettableFuture<T>> listeners = new HashSet<>();
 
+    @SuppressWarnings("CollectionUndefinedEquality")
```

Comment:
What’s this and why does it need to be suppressed? Can we fix the issue instead?

Reply:
https://github.com/google/error-prone/commit/1046226c77e28970f6ad2392a60cd9bb98a4911a

Reply:
> Warn on futureA.equals(futureB) (and popular subtypes). Futures do not have well-defined equality semantics.

Reply:
This check now applies to:

* "Future",
* "java.util.concurrent.Future",
* "java.util.concurrent.CompletableFuture",
* "com.google.common.util.concurrent.ListenableFuture",
* "com.google.common.util.concurrent.SettableFuture"

Reply:
Added a reference wrapper